File: src/main/java/net/Chidoziealways/everythingjapanese/item/ModEquipmentAssets.java
package net.Chidoziealways.everythingjapanese.item;

import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.minecraft.core.Registry;
import net.minecraft.resources.ResourceKey;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.equipment.EquipmentAsset;
import net.minecraft.world.item.equipment.EquipmentAssets;


public interface ModEquipmentAssets {
    ResourceKey<EquipmentAsset> PYRITE = createId("pyrite");
    ResourceKey<EquipmentAsset> NEPHRITE = createId("nephrite");
    ResourceKey<EquipmentAsset> SAMURAI = createId("samurai");

    static ResourceKey<EquipmentAsset> createId(String pName) {
        return ResourceKey.create(EquipmentAssets.ROOT_ID, ResourceLocation.fromNamespaceAndPath(EverythingJapanese.MOD_ID, pName));
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/item/ModEquipmentAssets.kt
package net.Chidoziealways.everythingjapanese.item

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.minecraft.resources.ResourceKey
import net.minecraft.resources.Identifier
import net.minecraft.world.item.equipment.EquipmentAsset
import net.minecraft.world.item.equipment.EquipmentAssets

interface ModEquipmentAssets {
    companion object {
        fun createId(pName: String): ResourceKey<EquipmentAsset> {
            return ResourceKey.create<EquipmentAsset>(
                EquipmentAssets.ROOT_ID,
                Identifier.fromNamespaceAndPath(JAPANESE_MOD_ID, pName)
            )
        }

        val PYRITE: ResourceKey<EquipmentAsset> = createId("pyrite")
        val NEPHRITE: ResourceKey<EquipmentAsset> = createId("nephrite")
        val SAMURAI: ResourceKey<EquipmentAsset> = createId("samurai")
    }
}
